lib/actions/apiV1.js

```diff
@@ -0,0 +1,233 @@
+import L from 'leaflet'
+import qs from 'qs'
+
+import {
+  bikeRentalQuery,
+  createQueryAction,
+  findGeometryForPattern,
+  findGeometryForTrip,
+  findNearbyAmenitiesError,
+  findNearbyAmenitiesResponse,
+  findNearbyStops,
+  findPatternsForRouteError,
+  findPatternsForRouteResponse,
+  findRouteError,
+  findRouteResponse,
+  findRoutesAtStop,
+  findStopError,
+  findStopResponse,
+  findStopsForTrip,
+  findStopTimesForStop,
+  findStopTimesForTrip,
+  findTripError,
+  findTripResponse,
+  parkAndRideQuery,
+  receivedVehiclePositions,
+  receivedVehiclePositionsError
+} from './api'
+import { setViewedStop } from './ui'
+import { zoomToStop } from './map'
+
+const findTrip = (params) =>
+  createQueryAction(
+    `index/trips/${params.tripId}`,
+    findTripResponse,
+    findTripError,
+    {
+      noThrottle: true,
+      postprocess: (payload, dispatch) => {
+        dispatch(findStopsForTrip({ tripId: params.tripId }))
+        dispatch(findStopTimesForTrip({ tripId: params.tripId }))
+        dispatch(findGeometryForTrip({ tripId: params.tripId }))
+      }
+    }
+  )
+
+export function vehicleRentalQuery(
+  params,
+  responseAction,
+  errorAction,
+  options
+) {
+  const paramsString = qs.stringify(params)
+  const endpoint = `vehicle_rental${paramsString ? `?${paramsString}` : ''}`
+  return createQueryAction(endpoint, responseAction, errorAction, options)
+}
+
+function findNearbyAmenities(params, stopId) {
+  return function (dispatch, getState) {
+    const { lat, lon, radius } = params
+    const bounds = L.latLng(lat, lon).toBounds(radius)
+    const { lat: low, lng: left } = bounds.getSouthWest()
+    const { lat: up, lng: right } = bounds.getNorthEast()
+    dispatch(
+      parkAndRideQuery(
+        { lowerLeft: `${low},${left}`, upperRight: `${up},${right}` },
+        findNearbyAmenitiesResponse,
+        findNearbyAmenitiesError,
+        {
+          noThrottle: true,
+          rewritePayload: (payload) => {
+            return {
+              parkAndRideLocations: payload,
+              stopId
+            }
+          }
+        }
+      )
+    )
+    dispatch(
+      bikeRentalQuery(
+        { lowerLeft: `${low},${left}`, upperRight: `${up},${right}` },
+        findNearbyAmenitiesResponse,
+        findNearbyAmenitiesError,
+        {
+          rewritePayload: (payload) => {
+            return {
+              bikeRental: payload,
+              stopId
+            }
+          }
+        }
+      )
+    )
+    dispatch(
+      vehicleRentalQuery(
+        { lowerLeft: `${low},${left}`, upperRight: `${up},${right}` },
+        findNearbyAmenitiesResponse,
+        findNearbyAmenitiesError,
+        {
+          rewritePayload: (payload) => {
+            return {
+              stopId,
+              vehicleRental: payload
+            }
+          }
+        }
+      )
+    )
+  }
+}
+
+export function findStop(params) {
+  return createQueryAction(
```

Comment:
I kinda like the `const ... => createQueryAction(...)` notation (multiple instances).

Reply:
Me too!

Reply:
Refactored in d2507aa982ecca23a82fb1964c5e7a7099b81d88